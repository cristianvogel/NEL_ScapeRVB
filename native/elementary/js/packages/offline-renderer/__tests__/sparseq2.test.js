import OfflineRenderer from '../index';
import { el } from '@elemaudio/core';


test('sparseq2 basics', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 0,
    numOutputChannels: 1,
  });

  // Graph
  core.render(
    el.sparseq2({seq: [
      { time: 512 * 10 + 4, value: 1 },
      { time: 512 * 10 + 8, value: 2 },
      { time: 512 * 10 + 12, value: 3 },
      { time: 512 * 10 + 16, value: 4 },
    ]}, el.time())
  );

  // Ten blocks of data
  let inps = [];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Push another small block. We should see the sequence play out
  inps = [];
  outs = [new Float32Array(32)];

  core.process(inps, outs);
  expect(outs[0]).toMatchSnapshot();
});

test('sparseq2 interp', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 0,
    numOutputChannels: 1,
  });

  // Graph
  core.render(
    el.sparseq2({
      interpolate: 1,
      seq: [
        { time: 512 * 10 + 0, value: 1 },
        { time: 512 * 10 + 4, value: 2 },
        { time: 512 * 10 + 8, value: 3 },
        { time: 512 * 10 + 12, value: 4 },
      ],
    }, el.time())
  );

  // Ten blocks of data
  let inps = [];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Push another small block. We should see the sequence play out with
  // intermediate interpolated values
  inps = [];
  outs = [new Float32Array(32)];

  core.process(inps, outs);
  expect(outs[0]).toMatchSnapshot();
});

test('sparseq2 looping', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 0,
    numOutputChannels: 1,
  });

  // Graph
  let loop = (start, end, t) => el.add(start, el.mod(t, el.sub(end, start)));

  core.render(
    el.sparseq2({seq: [
      { time: 512 * 10 + 0, value: 1 },
      { time: 512 * 10 + 4, value: 2 },
      { time: 512 * 10 + 8, value: 3 },
      { time: 512 * 10 + 12, value: 4 },
    ]}, loop(5120, 5120 + 16, el.time()))
  );

  // Ten blocks of data
  let inps = [];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Push another small block. We should see a loop through the sequence
  // on a 16 sample interval
  inps = [];
  outs = [new Float32Array(32)];

  core.process(inps, outs);
  expect(outs[0]).toMatchSnapshot();
});

test('sparseq2 skip ahead', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 1,
    numOutputChannels: 1,
  });

  // Graph
  let loop = (start, end, t) => el.add(start, el.mod(t, el.sub(end, start)));

  core.render(
    el.sparseq2({seq: [
      { time: 512 * 10 + 0, value: 1 },
      { time: 512 * 10 + 4, value: 2 },
      { time: 512 * 10 + 8, value: 3 },
      { time: 512 * 10 + 12, value: 4 },
    ]}, el.in({channel: 0}))
  );

  // Ten blocks of data
  let inps = [new Float32Array(512 * 10)];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Push another small block. We should see the sequence skip ahead
  // to a future value
  inps = [Float32Array.from([
    5120, 5120, 5120, 5120, 5120, 5120, 5120, 5120,
    5128, 5128, 5128, 5128, 5128, 5128, 5128, 5128,
  ])];

  outs = [new Float32Array(16)];

  core.process(inps, outs);
  expect(outs[0]).toMatchSnapshot();
});
