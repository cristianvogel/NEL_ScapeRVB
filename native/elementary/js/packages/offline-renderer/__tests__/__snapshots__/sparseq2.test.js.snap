// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sparseq2 basics 1`] = `
Float32Array [
  0,
  0,
  0,
  0,
  1,
  1,
  1,
  1,
  2,
  2,
  2,
  2,
  3,
  3,
  3,
  3,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
]
`;

exports[`sparseq2 interp 1`] = `
Float32Array [
  1,
  1.25,
  1.5,
  1.75,
  2,
  2.25,
  2.5,
  2.75,
  3,
  3.25,
  3.5,
  3.75,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
]
`;

exports[`sparseq2 looping 1`] = `
Float32Array [
  1,
  1,
  1,
  1,
  2,
  2,
  2,
  2,
  3,
  3,
  3,
  3,
  4,
  4,
  4,
  4,
  1,
  1,
  1,
  1,
  2,
  2,
  2,
  2,
  3,
  3,
  3,
  3,
  4,
  4,
  4,
  4,
]
`;

exports[`sparseq2 skip ahead 1`] = `
Float32Array [
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  1,
  3,
  3,
  3,
  3,
  3,
  3,
  3,
  3,
]
`;
